Readme.md

```diff
@@ -1,25 +1,87 @@
 # Flat files decoder for firehose
 
-## Usage
+[![CI status](https://github.com/semiotic-ai/flat-files-decoder/workflows/ci/badge.svg)][gh-ci]
+
+<!-- TODO: Seve please checkout if what I wrote makes sense -->
+this crate is designed to decompress and decode headers from [binary files, which are called flat files,](https://github.com/streamingfast/firehose-ethereum/blob/develop/proto/sf/ethereum/type/v2/type.proto) generated from Firehose. Flat files store all information necessary to reconstruct the transaction and receipt tries. It also checks the validity of 
+receipt roots and transaction roots present in the block headers by recalculating them via the block body data. Details of the implementation can be found [here](https://github.com/streamingfast/dbin?tab=readme-ov-file)
+
+This tool was first presented as a mean to enhance the performance and verifiability of The Graph protocol. However,
+it turns out it could be used as a solution for EIP-4444 problem of full nodes stopping to provide historical data over one year.
+The idea is that the flat files that this crate can decode could also be used as an archival format similar to era1 files, specially
+if they can be verified. 
```

Comment:
It might be worth adding a line about how we verify:

"See our ethresear.ch post, [here](https://ethresear.ch/t/using-the-graph-to-preserve-historical-data-and-enable-eip-4444/), and our pre-merge solution, implemented [here](https://github.com/semiotic-ai/header_accumulator) to see how we verify that the flat files are consistent with Ethereum's canonical history.